onnxruntime/core/providers/cuda/tensor/gather_elements.cc

```diff
@@ -9,115 +9,94 @@
 namespace onnxruntime {
 namespace cuda {
 
+// Ideally both input and indices can support strided tensor, for training case, the indices is the input for both
+// GatherElements and GatherElementsGrad, indices supporting strided tensor is more useful for saving memory.
+// So we only mark indices as MayStridedInput for now. Will do this for input once needed.
+#ifdef ENABLE_TRAINING
```

Comment:
I am wondering whether we should create a separate macro for enabling strided tensors... I believe the reason we are restricting this to training is because there were binary size implications for mobile builds right? Having this under an independent macro will make it easy for testing this for inference in future. The binary size restrictions are not so strict for other inference builds. We can certainly access the tradeoffs (perf\mem vs binary size).

Reply:
I agree with you. We can do this just like what ENABLE_ATEN does, and can enable it for training and eager mode by default. I am thinking we can do this (as well as the SupportStridedOutput renaming) in a new PR as it requires to change quite some other files not related to this.